### DISCUSSION TITLE
"Annotation matrix"

### DISCUSSION THREAD

**Discussion by rafaelkaufmann** [mancusolab/susiepca#18] [Q&A]
Hello! First of all, **thank you** for developing this library -- I've been integrating SuSiE-PCA into my project with great success and it is saving us several days of work and compute.

I have a question about the `A` function argument in `sp.infer.susie_pca`. The docstring says:

```
        A: Annotation matrix to use in parameterized-prior mode. If not `None`, leading dimension
            should match the feature dimension of X.
```

However, I can't find any explanation of either the concepts "annotation matrix" or "parameterized-prior mode" in the documentation or the paper. My first assumption was that it corresponds to a matrix of priors for the element-wise variances of W, which is referred to in the paper as $\sigma^2_0$. I tried to confirm this by reading the code, but there's a lot of indirection and it's unclear to me. There are also some oddities with this interpretation:

* It would require that $A_{ij} > 0 \ \forall i,j$, but there is no test for this and indeed I could successfully pass matrices that don't conform to this constraint to `susie_pca`.
* It would also require that `A.shape[1] == z_dim`, but `_check_args` only performs a check on `A.shape[0]`.

Another explanation is that A is a matrix of priors for the means of W. This would address the first problem, though not the second.

I've also found the term "annotation matrix" in other papers on PCA for bioinformatics, where it seems to correspond to a binary matrix which is 1 iff a causal relationship is known to exist between an a priori latent variable and an observable, and 0 otherwise. If this is the correct explanation, it would match my use case perfectly. The second problem is still there, though ;)

In any case, empirically, passing a matrix known to be close to the true W as A improves recovery of the true matrix quite a bit, so I must be on the right track :)

Thanks in advance for clarification!

**Comment by quattro** [comment]
Hi @rafaelkaufmann , apologies for the very late reply. The annotation matrix wasn't described in our original paper, but we briefly investigated its use for expanding performance. It describes `meta data` or `meta features` about the prior probabilities and fits a weighted logistic regression model.

The basic model assumes a fixed prior probability (pi_j) of the jth variable having a non-zero effect. The annotation matrix generalizes this to be `pi_j := softmax(A_j' theta)` where `A_j` describes a vector of some meta features for the jth variable and `theta` is the vector of prior log odds.
